pkg/adapter/nimbus-kyverno/manager/manager.go

```diff
@@ -177,16 +175,20 @@ func createOrUpdateKp(ctx context.Context, npName, npNamespace string) {
 				logger.Info("KyvernoPolicy created", "KyvernoPolicy.Name", kp.Name, "KyvernoPolicy.Namespace", kp.Namespace)
 			}
 		} else {
-			kp.ObjectMeta.ResourceVersion = existingKp.ObjectMeta.ResourceVersion
-			if err = k8sClient.Update(ctx, &kp); err != nil {
-				logger.Error(err, "failed to configure existing KyvernoPolicy", "KyvernoPolicy.Name", existingKp.Name, "KyvernoPolicy.Namespace", existingKp.Namespace)
-				return
+			if kpName != "" && kpName == kp.GetName() {
```

Comment:
Why do you want to restrict the updates to only this KP ? Are you getting a cycle of updates, and watch events ?

Reply:
To break the cycle, you can check if the existing kp is equal to the kp returned from the builder

Reply:
because the particular intent id comprises of multiple kyverno policies and if the single policy, for ex: mutateexisting one is getting updated then we should update only that particular policy not the mutateoncreate one.